WCFDataService/Client/System/Data/Services/Client/Binding/BindingGraph.cs

```diff
@@ -840,8 +840,8 @@ public void RemoveEdge(object source, object target, string label)
             /// <returns>true if an edge exists between source and target with given label, false otherwise</returns>
             public bool ExistsEdge(object source, object target, string label)
             {
-                Edge e = new Edge { Source = this.vertices[source], Target = this.vertices[target], Label = label };
-                return this.vertices[source].OutgoingEdges.Any(r => r.Equals(e));
+                Edge e = new Edge(this.vertices[source], this.vertices[target], label);
+                return this.vertices[source].OutgoingEdges.Contains(e);
```

Comment:
EdgeComparer GetHashCode is called automatically by the HashSet when an edge is added to the set. The returned hash code is used internally by the HashSet to check for equality - and thus also used from Contains. The HashCode for an edge is only calculated once during Add(). A HashCode for an object should be immutable - we don't want it to change over time - thats why I refactored the Edge class to ensure the Source, Target and Label cannot be changed from the outside.